src/charm.py

```diff
@@ -72,11 +74,20 @@ def __init__(self, *args):
         """Initialize defaults and event handlers."""
         super().__init__(*args)
 
+        self._database = DatabaseObserver(self)
+
+        self.framework.observe(
+            self._database.database.on.database_created, self._database_relation_changed
```

Comment:
Could you rename the methods according to the ISD014, please?

"In the __init__ of a CharmBase class, self.framework.observe is used to register functions that handle charm events. There should be a 1-1 mapping between events and observers, where the observer is named _on_<event-name> (for example, _on_config_changed)."

Reply:
Renamed